pallets/pips/src/lib.rs

```diff
@@ -1007,11 +1132,33 @@ decl_module! {
                 0
             })
         }
-
     }
 }
 
 impl<T: Trait> Module<T> {
+    /// Ensure that `origin` represents a signed extrinsic (i.e. transaction)
+    /// and confirms that the account is the same as the given `proposer`.
+    ///
+    /// For example, if `proposer` denotes a committee,
+    /// then `origin` is checked against the committee's origin.
+    ///
+    /// # Errors
+    /// * `BadOrigin` unless the checks above pass.
+    fn ensure_signed_by(origin: T::Origin, proposer: &Proposer<T::AccountId>) -> DispatchResult {
+        match proposer {
+            Proposer::Community(acc) => {
+                ensure!(acc == &ensure_signed(origin)?, DispatchError::BadOrigin)
+            }
+            Proposer::Committee(Committee::Technical) => {
```

Comment:
Is there any way to make this generic across committees (i.e. so the function doesn't need to be modified if we introduce a new committee)?

Reply:
Options:

1. Add `fn ensure_signed_committee(origin: T::Origin, committee: Committee) -> DispatchResult;` to the `trait Trait`, and move the logic here to where `Runtime` is defined.

   - pros: centralizes logic, removes 2 assoc types
   - con: still hard-coded at compile-time, the `Committee` enum is still defined here.

2. Refactor the committee module to handle committee instances at runtime instead of statically like now. Then we will register and manage a white-list of committees which may file PIPs. I figure this will require a bigger rewrite of the committee module. You'll also need a dynamic notion of a committees identity.

For now, I'd suggest leaving things as-is, since it wasn't clear there would be frequent additions of committees.

Reply:
Yep - that sounds fine. If we get into the situation where we are reguarly adding / removing committees we can amend logic then.